src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1052,6 +1052,35 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
 
     namespace ThreadNetworkDiagnostics = app::Clusters::ThreadNetworkDiagnostics;
 
+    if (otDatasetIsCommissioned(mOTInst))
+    {
+        // For the following nullable attributes of the cluster, encodeNull since
+        // thread instance cannot provide the related data when it is not currently configured.
+        //
+        // Note that RoutingRole is nullable but not listed here as thread provide
```

Comment:
Also, isn't RoutingRole determined dynamically in many cases, in which case it depends on not just having a dataset but in fact being attached?  I feel like this really needs to get clarified in the spec....

Reply:
RoutingRole also combines the state of the thread network. In a uncommissionned state, the device role/state would be detached, and when the radio is turned off it is disabled

The spec does define those two values in the RoutingRole enum, 

Disabled -> unspecified
Detached -> detached

Since they are valid values of the enum and maps to what the thread stack provides, I believe the nullable value isn't needed as then other values provide more information

Reply:
Ah, I see.  Great!